lib/identity_cache/query_api.rb

```diff
@@ -125,7 +125,7 @@ def get_embedded_association(record, association, options) #:nodoc:
         embedded_variable = record.instance_variable_get(:"@#{options[:records_variable_name]}")
         if IdentityCache.unmap_cached_nil_for(embedded_variable).nil?
           nil
-        elsif record.class.reflect_on_association(association).collection?
+        elsif embedded_variable.is_a?(Array)
           embedded_variable.map {|e| coder_from_record(e) }
         else
           coder_from_record(embedded_variable)
```

Comment:
Correct me if I'm wrong (I'm no expert here), but IDC does not have the gem version in it's cache key.

By removing `record.class.reflect_on_association(association).collection?`, you'll end up with a `embedded_variable` that will be a `ActiveRecord::Associations::CollectionProxy` from existing cache before the change and `coder_from_record` will not like it.